structify/structify_test.go

```diff
@@ -0,0 +1,154 @@
+package structify
+
+import (
+	"github.com/icinga/icinga-go-library/types"
+	"github.com/stretchr/testify/require"
+	"reflect"
+	"testing"
+)
+
+type initerTest = struct {
+	S string `test:"s"`
+	I int8   `test:"i"`
+}
+
+type stringTest = struct {
+	S string `test:"s"`
+}
+
+func testIniter(p any) {
+	p.(*initerTest).I = 42
+}
+
+func TestMakeMapStructifier(t *testing.T) {
+	subtests := []struct {
+		name   string
+		tag    string
+		initer func(any)
+		input  map[string]any
+		error  bool
+		output any
+	}{
+		{"empty", "", nil, nil, false, &struct{}{}},
```

Comment:
This wall of code is barely readable, and testing is about what is actually being tested, so please make it readable instead of always trying to save lines at all costs.